src/units/abilities.cpp

```diff
@@ -1481,20 +1525,53 @@ bool attack_type::special_active_impl(const_attack_ptr self_attack, const_attack
 	temporary_facing other_facing(other, other_loc.get_relative_dir(self_loc));
 
 	// Filter poison, plague, drain, first strike
-	if (tag_name == "drains" && other && other->get_state("undrainable")) {
+	if (tag_name == "drains" && other && (other->get_state("undrainable")|| other->get_ability_bool_anti_weapons("undrains", other_loc, !is_attacker, other_attack, self_attack))) {
 		return false;
 	}
 	if (tag_name == "plague" && other &&
 		(other->get_state("unplagueable") ||
+		other->get_ability_bool_anti_weapons("unplague", other_loc, !is_attacker, other_attack, self_attack) ||
 		 resources::gameboard->map().is_village(other_loc))) {
 		return false;
 	}
 	if (tag_name == "poison" && other &&
-		(other->get_state("unpoisonable") || other->get_state(unit::STATE_POISONED))) {
+		(other->get_state("unpoisonable") || other->get_ability_bool_anti_weapons("unpoison", other_loc, !is_attacker, other_attack, self_attack) || other->get_state(unit::STATE_POISONED))) {
+		return false;
+	}
+	if (tag_name == "slow" && other &&
+		other->get_ability_bool_anti_weapons("unslow", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "petrifies" && other &&
+		other->get_ability_bool_anti_weapons("unpetrifies", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "attacks" && other &&
+		other->get_ability_bool_anti_weapons("unattacks", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "damage" && other &&
+		other->get_ability_bool_anti_weapons("undamage", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "chance_to_hit" && other &&
+		other->get_ability_bool_anti_weapons("unchance_to_hit", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "swarm" && other &&
+		other->get_ability_bool_anti_weapons("unswarm", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "heal_on_hit" && other &&
+		other->get_ability_bool_anti_weapons("unheal_on_hit", other_loc, !is_attacker, other_attack, self_attack)) {
+		return false;
+	}
+	if (tag_name == "berserk" && other &&
+		other->get_ability_bool_anti_weapons("unberserk", other_loc, !is_attacker, other_attack, self_attack)) {
 		return false;
 	}
-	if (tag_name == "firststrike" && !is_attacker && other_attack &&
-		other_attack->has_special_or_ability("firststrike")) {
+	if (tag_name == "firststrike" && !is_attacker && ((other_attack &&
+		other_attack->has_special_or_ability("firststrike")) || (other && other->get_ability_bool_anti_weapons("unfirststrike", other_loc, !is_attacker, other_attack, self_attack)))) {
```

Comment:
I haven't checked if the called functions handle this, but what happens when:
* attacker has firststrike
* defender has firststrike and unfirststrike

I think that should let the defender strike first, but think it's going to fail on the `other_attack->has_special_or_ability("firststrike")` call.

Reply:
if defender and attacker has firsttrike then, the fight played like if none has firststrike, and with unfirsttsrike ability, it's like attacker don't have special, same if defender don't have firststike